gateway/src/apicast/policy/rewrite_url_captures/named_args_matcher.lua

```diff
@@ -0,0 +1,140 @@
+local re_gsub = ngx.re.gsub
+local re_match = ngx.re.match
+local re_gmatch = ngx.re.gmatch
+local re_split = require('ngx.re').split
+local insert = table.insert
+local format = string.format
+local ipairs = ipairs
+local setmetatable = setmetatable
+
+local _M = {}
+
+local mt = { __index = _M }
+
+local function split(string, separator, max_matches)
+  return re_split(string, separator, 'oj', nil, max_matches)
+end
+
+-- Returns a list of named args extracted from a match_rule.
+-- For example, for the rule /{abc}/{def}?{ghi}=1, it returns this list:
```

Comment:
So, do you think it makes sense to use just `{}` instead of `{{ }}` to match with the liquid syntax?

Reply:
I think `{}` is better in this case because it matches the same format we are using for the mapping rules.

Reply:
Ok, thats fair point. And for the replacement ? Shouldn't we just pass it through liquid?

Reply:
Not sure. That would only replace the code we have in `replace_in_template` which is not problematic, in my opinion.

Reply:
Yep. I'm thinking about consistency. If push people everywhere to use liquid, then it would make sense to stick to it. I have no hard feelings one way or the other, just want to avoid confused people that can't use what they do everywhere else.

But it will not be hard to migrate. Just adding the `template_type`. So if you have no feelings one way or the other then lets merge this 👍

Reply:
I'd rather leave this one as it is, at least for now :+1: